pulsar-discovery-service/src/main/java/org/apache/pulsar/discovery/service/server/DiscoveryServiceStarter.java

```diff
@@ -40,6 +40,15 @@
  */
 public class DiscoveryServiceStarter {
 
+    public static void checkConfig(ServiceConfig config) {
+        checkArgument(!isEmpty(config.getZookeeperServers()), "zookeeperServers must be provided");
+        if(!isEmpty(config.getGlobalZookeeperServers()) && isEmpty(config.getConfigurationStoreServers())) {
```

Comment:
Why doesn't ServiceConfig handle this in the same way as ServiceConfiguration?

Reply:
Oh, it seems it does. This snippet here is unneeded.

Reply:
Thanks, would like to keep it to do a double checking.

Reply:
This will never evaluate to true though, so you'd be better off putting it in a checkArgument.
```
checkArgument(!isEmpty(config.getConfigurationStoreServers()) || isEmpty(config.getGlobalZookeeperServers()))
```